Chapter15/proxy.erl

```diff
@@ -0,0 +1,36 @@
+-module(proxy).
+-export([server/0, wait_connect/2, get_request/2]).
+
+
+server() ->
+  {ok, ListenSocket} = gen_tcp:listen(1501, [binary, {active, false}]),
+  wait_connect(ListenSocket,0).
+
+wait_connect(ListenSocket, Count) ->
+  {ok, Socket} = gen_tcp:accept(ListenSocket),
+  spawn(?MODULE, wait_connect, [ListenSocket, Count+1]),
+  get_request(Socket, Count).
+
+get_request(Socket, Count) ->
+  case gen_tcp:recv(Socket, 0, 5000) of
+    {ok, Binary} ->
+      handle([Binary], Count, Socket);
+    {error, closed} ->
+      ok
+  end.
+
+% the http module is no longer used and I could not get the newer httpc module
+% to return the site in the same format as the http module would have 
+
+handle([<<_Get:4/binary, Tail/binary>>], _Count, Socket) ->
+  Url = extract_url(Tail, []),
+  % {ok, Site} = httpc:request(Url),
```

Comment:
shouldn't `httpc:request/1` do it?
Something like:
```erlang
{ok, {{_HttpVersion, 200, "OK"}, _Headers, _Body}} = httpc:request("http://erlang.org" ++ Url).
```

Reply:
All fixed